core/capabilities/ccip/ccipevm/pluginconfig.go

```diff
@@ -0,0 +1,119 @@
+)
+
+// PluginConfig is a struct that contains the configuration for a plugin.
+type PluginConfig struct {
+	extraDataCodec ccipcommon.ExtraDataCodec
+}
+
+// NewPluginConfig returns a new PluginConfig.
+func NewPluginConfig(extraDataCodec ccipcommon.ExtraDataCodec) *PluginConfig {
+	return &PluginConfig{
+		extraDataCodec: extraDataCodec,
+	}
+}
+
+// InitializePluginConfig returns a PluginConfig for EVM chains.
+func (p PluginConfig) InitializePluginConfig() ccipcommon.PluginConfig {
+	return ccipcommon.PluginConfig{
+		CommitPluginCodec:  NewCommitPluginCodecV1(),
+		ExecutePluginCodec: NewExecutePluginCodecV1(p.extraDataCodec),
+		MessageHasher: func(lggr logger.Logger) cciptypes.MessageHasher {
+			return NewMessageHasherV1(lggr, p.extraDataCodec)
+		},
+		TokenDataEncoder:     NewEVMTokenDataEncoder(),
+		GasEstimateProvider:  NewGasEstimateProvider(),
+		RMNCrypto:            func(lggr logger.Logger) cciptypes.RMNCrypto { return NewEVMRMNCrypto(lggr) },
+		GetChainReaderWriter: GetCRCW{},
+	}
+}
+
+// GetCRCW is a struct that implements the GetChainReaderWriter interface for EVM chains.
+type GetCRCW struct{}
+
+// GetChainReader returns a new ContractReader for EVM chains.
+func (g GetCRCW) GetChainReader(ctx context.Context, params ccipcommon.GetChainReaderParams) (types.ContractReader, error) {
+	var chainReaderConfig evmrelaytypes.ChainReaderConfig
+	if params.ChainID == params.DestChainID {
+		chainReaderConfig = evmconfig.DestReaderConfig
+	} else {
+		chainReaderConfig = evmconfig.SourceReaderConfig
+	}
+
+	if !params.Ofc.CommitEmpty() && params.Ofc.Commit().PriceFeedChainSelector == params.ChainSelector {
+		params.Lggr.Debugw("Adding feed reader config", "chainID", params.ChainID)
+		chainReaderConfig = evmconfig.MergeReaderConfigs(chainReaderConfig, evmconfig.FeedReaderConfig)
+	}
+
+	if isUSDCEnabled(params.Ofc) {
+		params.Lggr.Debugw("Adding USDC reader config", "chainID", params.ChainID)
+		chainReaderConfig = evmconfig.MergeReaderConfigs(chainReaderConfig, evmconfig.USDCReaderConfig)
+	}
+
+	if params.ChainID == params.HomeChainID {
+		params.Lggr.Debugw("Adding home chain reader config", "chainID", params.ChainID)
+		chainReaderConfig = evmconfig.MergeReaderConfigs(chainReaderConfig, evmconfig.HomeChainReaderConfigRaw)
+	}
+
+	marshaledConfig, err := json.Marshal(chainReaderConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal chain reader config: %w", err)
+	}
+
+	cr, err := params.Relayer.NewContractReader(ctx, marshaledConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return cr, nil
+}
+
+// GetChainWriter returns a new ContractWriter for EVM chains.
+func (g GetCRCW) GetChainWriter(ctx context.Context, params ccipcommon.GetChainWriterParams) (types.ContractWriter, error) {
+	var fromAddress common.Address
+	transmitter, ok := params.Transmitters[types.NewRelayID(params.ChainFamily, params.ChainID)]
+	if ok {
+		fromAddress = common.HexToAddress(transmitter[0])
+	}
```

Comment:
Not sure how this'd happen, but should we return an `error` if `!ok`?

Reply:
I believe this is due to constructing chain-writers in order to get gas prices for source chains. In that case there's no address because they aren't actually used for transmitting.